src/ExadelMentorship.Persistence/WeatherHistorySavingRepository.cs

```diff
@@ -14,15 +14,15 @@ public WeatherHistorySavingRepository(ApplicationDbContext context)
             _dbContext = context;
         }
 
-        public async Task SaveInDbAsync(string cityName, double temperature)
+        public void SaveInDb(string cityName, double temperature)
         {
-            await _dbContext.AddAsync(new WeatherHistory
+            _dbContext.Add(new WeatherHistory
             {
                 CityName = cityName,
                 Temperature = temperature,
                 Time = DateTime.Now
             });
-            await _dbContext.SaveChangesAsync();
+            _dbContext.SaveChanges();
```

Comment:
SaveChanges is IO-operation i.e. blocking operation. You better revert back to async/await

Reply:
done